daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/BinaryNumberUnparsers.scala

```diff
@@ -268,6 +317,10 @@ abstract class BinaryDecimalUnparserBase(
     nBits: Int,
     finfo: FormatInfo
   ): Boolean = {
-    dos.putBigInt(asBigInt(value), nBits, signed == YesNo.Yes, finfo)
+    if (nBits > 0) {
+      dos.putBigInt(asBigInt(value), nBits, signed == YesNo.Yes, finfo)
```

Comment:
Does BinaryDecimal have the same restriction about signed/unsigned and 1-bit/2-bit? For example, if `signed == Yes` and `nBits == 1`, should that be a warning or UE depending on the tunable value?

Reply:
I'm not certain since the ticket only specified Binary Integers. @mbeckerle  thoughts?

Reply:
type xs:decimal with binaryNumberRep "binary" is a signed integer that is then scaled by binaryDecimalVirtualPoint. 

Before the scaling, I think it behaves exactly like xs:integer. For signed, it must have 2 bits. Unsigned (which is what property dfdl:decimalSigned controls - despite the DFDL spec not providing adequate description of the property) means you must have at least 1 bit.